modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDynamicParameterTest.java

```diff
@@ -235,6 +236,23 @@ public void testWithDifferentParametersTypes() {
         assertQuery("SELECT UPPER(TYPEOF(?))").withParams(1d).returns("DOUBLE").check();
     }
 
+    /** Check that insertion value is trimmed if leading is zeroes. */
+    @ParameterizedTest
+    @CsvSource({"BINARY", "VARBINARY"})
+    public void testInsertZeroContainedBinary(String type) {
+        sql(format("CREATE TABLE t(id INT PRIMARY KEY, val {}(5000))", type));
```

Comment:
I would suggest to create table with 3 columns instead of creating 2 tables with 2 columns

Reply:
done